Lib/Inc/CGpioWrapper.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * CGpioWrapper.h
+ *
+ *  Created on: 14 Feb 2022
+ *      Author: salavat.magazov
+ */
+
+#ifndef CGPIOWRAPPER_H_
+#define CGPIOWRAPPER_H_
+
+#include "gpio.h"
+
+class CGpioWrapper
+{
+public:
+    CGpioWrapper(GPIO_TypeDef *p_port, uint32_t pin);
```

Comment:
If you're using the GPIO_PIN_X format, the max value is 0xFFFF (all pins), so no need for 32 bits. ```uint16_t``` should be enough.

Reply:
But the CubeMX defines pins as uint32_t, so if I use uint16_t the compiler is going to give warnings about mismatched variables and reduced scope.

Reply:
Turns out I was wrong that CubeMX defines them as 32 bit unsigned int. they are just #define values. I will change it.